classes/local/utils.php

```diff
@@ -0,0 +1,482 @@
+                if (!isset($flavors[$flavor])) {
+                    continue;
+                }
+                $flavors[$flavor]->categories[] = $component['compcat'];
+            }
+
+            // Add variants to components structure.
+            $components[$key]['variants'] = $componentvariants[$component['name']] ?? [];
+        }
+
+        foreach ($flavors as $flavor) {
+            $flavor->categories = join(',', array_unique($flavor->categories));
+        }
+
+        return [
+                'components' => $components,
+                'categories' => $compcats,
+                'flavors' => $flavors,
+                'variants' => $variants,
+        ];
+    }
+
+    /**
+     * Rebuild the css cache.
+     *
+     * @return int the new revision for the cache
+     */
+    public static function rebuild_css_cache(): int {
+        global $DB;
+        $cache = \cache::make('tiny_elements', constants::CACHE_AREA);
+        $iconcssentries = [];
+        $componentcssentries = [];
+        $variantscssentries = [];
+        $components = [];
+        $variants = [];
+        try {
+            $components = $DB->get_records('tiny_elements_component', null, '', 'id, name, css, iconurl');
+            $categorycssentries = $DB->get_fieldset('tiny_elements_compcat', 'css');
+            $flavors = $DB->get_records('tiny_elements_flavor', null, 'id, name');
+            $flavorcssentries = $DB->get_fieldset('tiny_elements_flavor', 'css');
+            $variants = $DB->get_records('tiny_elements_variant', null, '', 'name, iconurl, css');
+        } catch (\dml_exception $e) {
+            // This is done to prevent the plugin from crashing the whole site if the database tables
+            // are not yet installed for some reason.
+            return 0;
+        }
+        foreach ($variants as $variant) {
+            $variantscssentries[] = $variant->css;
+            if (empty($variant->iconurl)) {
+                continue;
+            }
+            $iconcssentries[] = self::variant_icon_css($variant->name, self::replace_pluginfile_urls($variant->iconurl, true));
+        }
+        $componentflavors = $DB->get_records('tiny_elements_comp_flavor');
+        foreach ($componentflavors as $componentflavor) {
+            if (empty($componentflavor->iconurl)) {
+                continue;
+            }
+            $iconcssentries[] .= self::button_icon_css(
+                $componentflavor->componentname,
+                self::replace_pluginfile_urls($componentflavor->iconurl, true),
+                $componentflavor->flavorname
+            );
+        }
+        foreach ($components as $component) {
+            $componentcssentries[] = $component->css;
+            if (empty($component->iconurl)) {
+                continue;
+            }
+            $iconcssentries[] .= self::button_icon_css($component->name, self::replace_pluginfile_urls($component->iconurl, true));
```

Comment:
```suggestion
            $iconcssentries[] = self::button_icon_css($component->name, self::replace_pluginfile_urls($component->iconurl, true));
```
Does this . have any special meaning? It apparently works anyway, but can't find the difference :)